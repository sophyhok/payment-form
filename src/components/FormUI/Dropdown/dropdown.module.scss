.dropdown {
  display: inline-flex;
  width: 90px;
  background-color: #FFFFFF;
  height: 32px;
  padding: 0 16px;
  border: 1px solid #939FB1;
  border-radius: 5px;
  box-sizing: border-box;
  color: #1D2A3C;

  &__boxStatus_danger {
    border: 1px solid #FF0000;
  }

  &__boxStatus_sucess {
    border: 1px solid #00FF00;
  }

  &__boxStatus_none {
    border: 1px solid #939FB1;
  }
}