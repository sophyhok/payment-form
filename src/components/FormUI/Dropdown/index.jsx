import React from 'react';
import { string, array, func, oneOf } from 'prop-types';
import classnames from 'classnames';

import styles from './dropdown.module.scss';

const propTypes = {
  name: string,
  options: array.isRequired,
  onChange: func,
  boxStatus: oneOf(['none','success','danger']),
  testId: string,
}

const defaultProps = {
  name: 'button',
  onChange: () => { },
  boxStatus: 'none',
  testId: 'dropdown',
}

const Dropdown = ({
  name,
  options,
  onChange,
  boxStatus,
  testId 
}) => {
  return (
      <select
        className={classnames(styles.dropdown, { [styles[`dropdown__boxStatus_${boxStatus}`]]: boxStatus })}
        name={name}
        onChange={onChange}
        data-testid={testId}
      >
        {options.map(item => {
          return (
            <option key={item}>{item}</option>
          )
        })}
      </select>
  )
}

Dropdown.propTypes = propTypes;
Dropdown.defaultProps = defaultProps;

export default Dropdown;