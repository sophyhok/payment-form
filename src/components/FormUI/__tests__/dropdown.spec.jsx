import React from 'react';
import { render } from '@testing-library/react';

import Dropdown from '../Dropdown';

const options = ['one','two','three'];

it('should match snapshot', () => {
  const { asFragment } = render(<Dropdown options={options} />)
  expect(asFragment()).toMatchSnapshot();
})

it('should return undefined when deafult onChange runs', () => {
  const dropdown = Dropdown.defaultProps.onChange();
  expect(dropdown).toBe(undefined);
})

it('should render boxStatus with danger when props passed in', () => {
  const { getByTestId } = render(<Dropdown options={options} boxStatus="danger" />);
  const dropdown = getByTestId('dropdown');
  expect(dropdown.className).toMatch('dropdown__boxStatus_danger');
});
