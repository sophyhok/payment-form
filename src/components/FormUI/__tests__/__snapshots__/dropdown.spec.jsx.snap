// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <select
    class="dropdown dropdown__boxStatus_none"
    data-testid="dropdown"
    name="button"
  >
    <option>
      one
    </option>
    <option>
      two
    </option>
    <option>
      three
    </option>
  </select>
</DocumentFragment>
`;
